tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -396,21 +397,39 @@ SpanBuffer SwapBuffers()
             }
 
             RunSpanSampler(spans);
-
+            int? chunkSamplingPriority = null;
             if (CanComputeStats)
             {
                 spans = _statsAggregator?.ProcessTrace(spans) ?? spans;
                 bool shouldSendTrace = _statsAggregator?.ShouldKeepTrace(spans) ?? true;
                 _statsAggregator?.AddRange(spans);
+                var singleSpanSamplingSpans = new List<Span>(); // TODO maybe we can store this from above?
+
+                for (var i = spans.Offset; i < spans.Count; i++)
```

Comment:
This loop doesn't look right. If the `ArraySegment` has `Offset=10` and `Count=2`, you would want `i` to be `10` first, then `11`:
```csharp
// start
spans.Array[10]
spans.Array[11]
// stop
```

However, the condition `i < spans.Count` will be false (`10 < 2`) right off the bat and the code inside the loop would never run.

I think this works today because all our `ArraySegment` have `Offset=0`, but would should not rely on that.

Reply:
Thanks for catching this!

I'll add a test that covers an `ArraySegment` without an `Offset` of `0` and update the loop

Reply:
Should be fixed in ae94c00cd438e733d9e1a0d472a77b0e9372c10d

I've changed one of the tests to have a non-zero Offset as well to test